nmi_mysql/nmi_mysql.py

```diff
@@ -1,194 +1,122 @@
+                multi_insert.append(self._to_string(param))
 
             else:
-                params.append(self.to_string(param))
+                other_queries.append(self._to_string(param))
 
-        if values:
-            params = ', '.join(values)
-            query = query % params[1:-1]
+        if multi_insert:
+            return (query % ', '.join(multi_insert), params)
 
         else:
-            query = query % tuple(params)
-
-        return query
+            return (query % tuple(other_queries), params)
 
-    def to_string(self, temp):
-        if isinstance(temp, (list, tuple)):
+    def _to_string(self, temp):
+        if isinstance(temp, list):
             tmp = []
             for item in temp:
-                if isinstance(item, str):
-                    item = item.replace('%', '%%')
-                tmp.append(self.handle.escape(item))
+                tmp.append(self._to_string(item))
 
             return ', '.join(tmp)
 
         elif isinstance(temp, dict):
             tmp = []
             for key in temp:
-                if isinstance(temp[key], str):
-                    temp[key] = temp[key].replace('%', '%%')
-                tmp.append(key + ' = ' + self.handle.escape(temp[key]))
+                tmp.append(key + ' = ' + self._to_string(temp[key]))
 
             return ', '.join(tmp)
 
-        elif isinstance(temp, str):
-            return self.handle.escape(temp.replace('%', '%%'))
-
-        else:
-            return self.handle.escape(temp)
-
+        return '%s'
 
-class ConnectionPool(object):
-    """
-    Usage:
-        conn_pool = nmi_mysql.ConnectionPool(config)
-
-        db = conn_pool.get_connection()
-        db.query('SELECT 1', [])
-        conn_pool.return_connection(db)
-
-        conn_pool.close()
-    """
-
-    def __init__(self, conf, max_pool_size=20):
-        self.conf = conf
-        self.max_pool_size = max_pool_size
-        self.initialize_pool()
+    def connect(self):
+        self.con = self.engine.connect()
 
-    def initialize_pool(self):
-        self.pool = Queue(maxsize=self.max_pool_size)
-        for _ in range(0, self.max_pool_size):
-            self.pool.put_nowait(DB(self.conf, True))
+    def close(self):
+        self.con.close()
 
-    def get_connection(self):
-        # returns a db instance when one is available else waits until one is
-        db = self.pool.get(True)
+    def query(self, _query, _params=None):
+        if not self.con:
+            self.connect()
 
-        # checks if db is still connected because db instance automatically
-        # closes when not in used
-        if not self.ping(db):
-            db.connect()
+        result = None
 
-        return db
+        (query, params) = self._generate_query(_query, _params)
 
-    def return_connection(self, db):
-        return self.pool.put_nowait(db)
+        try:
+            result = self.con.execute(query, *params)
```

Comment:
Can you add,
`self._latest_query` and `self._latest_param` variable?
Just to help in debugging.